DICOMPlugins/DICOMM3DPlugin.py

```diff
@@ -0,0 +1,641 @@
+    assert os.path.exists(stlFilePath)
+    file.close()
+
+    # Create Segmentation Node based on .STL file
+    self._createSegmentationNode(loadable, stlFilePath)
+    
+    self.cleanup()
+    
+    return True
+
+  def _createSegmentationNode(self, loadable, stlFilePath):
+    #create modelNode
+    modelNode = slicer.util.loadModel(stlFilePath)
+
+    # Initialize Segmentation Node
+    segmentationNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLSegmentationNode")
+    segmentationNode.SetName(loadable.name)
+
+    # Initialize Segmentation Display Node
+    segmentationDisplayNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLSegmentationDisplayNode")
+    segmentationNode.SetAndObserveDisplayNodeID(segmentationDisplayNode.GetID())
+
+    #Setup converters for further export to labelmap or model representations 
+    vtkSegConverter = vtkSegmentationCore.vtkSegmentationConverter
+    segmentation = vtkSegmentationCore.vtkSegmentation()
+    segmentation.SetMasterRepresentationName(vtkSegConverter.GetSegmentationBinaryLabelmapRepresentationName())
+    segmentation.CreateRepresentation(vtkSegConverter.GetSegmentationClosedSurfaceRepresentationName(), True)
+    segmentationNode.SetAndObserveSegmentation(segmentation)
+    
+    #Load Model Node into Segmentation Node
+    self._importModelToSegAndRemoveModel(modelNode, segmentationNode)
+    self.addSeriesInSubjectHierarchy(loadable, segmentationNode)
+    return segmentationNode
+  
+  def _importModelToSegAndRemoveModel(self, ModelNode, segmentationNode):
+    segmentationsLogic = slicer.modules.segmentations.logic()
+    segmentation = segmentationNode.GetSegmentation()
+    numberOfSegmentsBeforeImport = segmentation.GetNumberOfSegments()
+    success = segmentationsLogic.ImportModelToSegmentationNode(ModelNode, segmentationNode)
+    if segmentation.GetNumberOfSegments() == 0:
+        logging.warning("Empty segment loaded from DICOM SEG!")
+    if segmentation.GetNumberOfSegments() - numberOfSegmentsBeforeImport > 1:
+        logging.warning("Multiple segments were loaded from DICOM SEG labelmap. Only one label was expected.")
+    if success and segmentation.GetNumberOfSegments()>0:
+      segment = segmentation.GetNthSegment(segmentation.GetNumberOfSegments() - 1)
+      segment.SetName("Segment 1")
+      segment.SetNameAutoGenerated(False)
+      self._removeModelNode(ModelNode)
+    return segmentation
+  
+  def _removeModelNode(self, modelNode):
+    #Model Node being temporary, we remove here the node from the scene 
+    dNode = modelNode.GetDisplayNode()
+    if dNode is not None:
+      slicer.mrmlScene.RemoveNode(dNode)
+    slicer.mrmlScene.RemoveNode(modelNode)
+
+  def examineForExport(self, subjectHierarchyItemID):
+    shNode = slicer.vtkMRMLSubjectHierarchyNode.GetSubjectHierarchyNode(slicer.mrmlScene)
+
+    exportable = self._examineExportableForSegmentationNode(shNode, subjectHierarchyItemID)
+    return self._setupExportable(exportable, subjectHierarchyItemID)
```

Comment:
This doesn't appear to export as M3D.  Is this just a duplicate of the code from the segmentation plugin?  If so it should be removed.  I don't remember the exact method but I believe that if your plugin doesn't support exporting you either project a stub of an exporter class or leave it out completely.